org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/CloseStartTagCodeAction.java

```diff
@@ -0,0 +1,153 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.contentmodel.participants.codeactions;
+
+import java.util.List;
+
+import org.eclipse.lemminx.commons.BadLocationException;
+import org.eclipse.lemminx.commons.CodeActionFactory;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.dom.DOMElement;
+import org.eclipse.lemminx.dom.DOMNode;
+import org.eclipse.lemminx.dom.LineIndentInfo;
+import org.eclipse.lemminx.services.extensions.ICodeActionParticipant;
+import org.eclipse.lemminx.services.extensions.IComponentProvider;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lemminx.utils.XMLPositionUtility;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+
+/**
+ * Code action to fix close start tag element.
+ *
+ */
+public class CloseStartTagCodeAction implements ICodeActionParticipant {
+
+	@Override
+	public void doCodeAction(Diagnostic diagnostic, Range range, DOMDocument document, List<CodeAction> codeActions,
+			SharedSettings sharedSettings, IComponentProvider componentProvider) {
+		Range diagnosticRange = diagnostic.getRange();
+		try {
+			int startOffset = document.offsetAt(diagnosticRange.getStart());
+			DOMNode node = document.findNodeAt(startOffset);
+			if (node != null && node.isElement()) {
+				int diagnosticEndOffset = startOffset
+						+ (diagnosticRange.getEnd().getCharacter() - diagnosticRange.getStart().getCharacter());
+				DOMElement element = (DOMElement) node;
+				if (!element.hasStartTag()) {
+					// </foo>
+					DOMElement parent = element.getParentElement();
+					if (parent != null && parent.getTagName() != null) {
+						// <a><b></c>
+						// Replace "c" with "b"
+						String tagName = parent.getTagName();
+						Range replaceRange = XMLPositionUtility.selectEndTagName(element);
+						CodeAction replaceTagAction = CodeActionFactory.replace("Replace with '" + tagName + "' tag",

```

Comment:
I think it might make more sense to the user if the message was "Replace with 'b' closing tag"

Reply:
I agree, fixed